src/main/java/org/reaktivity/nukleus/socks/internal/stream/SocksServerFactory.java

```diff
@@ -0,0 +1,801 @@
+            long routeId,
+            long initialId,
+            long replyId)
+        {
+            this.network = network;
+            this.routeId = routeId;
+            this.initialId = initialId;
+            this.replyId = replyId;
+            this.decodeState = this :: decodeHandshake;
+        }
+
+        private int decodeCommand(
+            DirectBuffer buffer,
+            int offset,
+            int limit)
+        {
+            SocksCommandRequestFW commandRequest = socksCommandRequestRO.tryWrap(buffer, offset, limit);
+            int progress = offset;
+            if (commandRequest != null)
+            {
+                progress = commandRequest.limit();
+                onSocksCommandRequest(commandRequest);
+            }
+            return progress;
+        }
+
+        private int decodeHandshake(
+            DirectBuffer buffer,
+            int offset,
+            int limit)
+        {
+            SocksHandshakeRequestFW handshakeRequest = handshakeRequestRO.tryWrap(buffer, offset, limit);
+            int progress = offset;
+            if (handshakeRequest != null)
+            {
+                progress = handshakeRequest.limit();
+                onHandshakeRequest(handshakeRequest);
+            }
+            return progress;
+        }
+
+        private void onNetwork(
+            int msgTypeId,
+            DirectBuffer buffer,
+            int index,
+            int length)
+        {
+            switch (msgTypeId)
+            {
+                case BeginFW.TYPE_ID:
+                    final BeginFW begin = beginRO.wrap(buffer, index, index + length);
+                    onNetworkBegin(begin);
+                    break;
+                case DataFW.TYPE_ID:
+                    final DataFW data = dataRO.wrap(buffer, index, index + length);
+                    onNetworkData(data);
+                    break;
+                case EndFW.TYPE_ID:
+                    final EndFW end = endRO.wrap(buffer, index, index + length);
+                    onNetworkEnd(end);
+                    break;
+                case AbortFW.TYPE_ID:
+                    final AbortFW abort = abortRO.wrap(buffer, index, index + length);
+                    onNetworkAbort(abort);
+                    break;
+                case WindowFW.TYPE_ID:
+                    final WindowFW window = windowRO.wrap(buffer, index, index + length);
+                    onNetworkWindow(window);
+                    break;
+                case ResetFW.TYPE_ID:
+                    final ResetFW reset = resetRO.wrap(buffer, index, index + length);
+                    onNetworkReset(reset);
+                    break;
+            }
+        }
+
+        private void onNetworkData(
+            DataFW data)
+        {
+            final OctetsFW payload = data.payload();
+            initialBudget -= Math.max(data.length(), 0) + data.padding();
+            if (initialBudget < 0)
+            {
+                //TODO
```

Comment:
Please add the `RESET` to reject the stream when over budget.
You will also need to `ABORT` the target stream if handshake and connect are complete.